src/js/components/Form/index.d.ts

```diff
@@ -10,13 +10,13 @@ export interface FormProps<T> {
   infos?: {};
   messages?: {invalid?: string,required?: string};
   onChange?: (value: T) => void;
-  onSubmit?: ((event: FormExtendedEvent) => void);
+  onSubmit?: ((event: FormExtendedEvent<T>) => void);
```

Comment:
this looks good to me. It's maybe a little weird since the type param naming here T is referring to R above...but maybe it isn't that weird and its nbd either way. 

Good addition!